### PR TITLE
Decompile / port fldeff_80F9BCC

### PR DESCRIPTION
**Pull request by garakmon**
2 NONMATCHING currently -- ~~will try to fix soon~~ I can't match them

### CODE REVIEW

File: src/battle_setup.c [hunk 1]
```diff
@@ -338,7 +338,7 @@ static void Task_BattleStart(u8 taskId)
     switch (tState)
     {
     case 0:
-        if (!FieldPoisonEffectIsRunning()) // is poison not active?
+        if (!FldeffPoison_IsActive()) // is poison not active?
```
Comment (Diegoisawesome) [inline src/battle_setup.c]: FldEff

File: src/fldeff_80F9BCC.c [hunk 2]
```diff
@@ -0,0 +1,1560 @@
...
+
+void sub_80F9C44(void (*taskfunc) (u8), u16 a1, u16 a2, u8 a3)
+{
+    u16 tempA, tempB;
+
+    u8 taskId = CreateTask(taskfunc, a3);
+
+    gTasks[taskId].data[0] = 0;
+
+    tempA = 16;
```
Comment (Diegoisawesome) [inline src/fldeff_80F9BCC.c]: Pretty sure these are ternaries

File: src/fldeff_80F9BCC.c [hunk 3]
```diff
@@ -0,0 +1,1560 @@
...
+        if (task->data[5] > 0x4F || task->data[6] < 0x52)
+        {
+            task->data[5] = 0x50;
+            task->data[6] = 0x51;
+            SetGpuReg(REG_OFFSET_BLDCNT, 0xBF);
+            SetGpuReg(REG_OFFSET_BLDY, 0x10);
+        }
+        SetGpuReg(REG_OFFSET_WIN0V, WIN_RANGE(task->data[5], task->data[6]));
+
+        if (task->data[5] != 0x50) return;
```
Comment (Diegoisawesome) [inline src/fldeff_80F9BCC.c]: Put returns on newlines and remove newlines after breaks

File: src/fldeff_80F9BCC.c [hunk 4]
```diff
@@ -0,0 +1,1560 @@
...
+    gFieldEffectArguments[6] = y;
+
+    switch (GetPlayerFacingDirection())
+    {
+    case DIR_SOUTH:
+        CreateSprite(&gUnknown_0858E68C,
+                     gSprites[gPlayerAvatar.spriteId].oam.x + 8,
+                     gSprites[gPlayerAvatar.spriteId].oam.y + 32,
+                     0);
+
```
Comment (Diegoisawesome) [inline src/fldeff_80F9BCC.c]: Remove extra newlines

File: src/fldeff_80F9BCC.c [hunk 5]
```diff
@@ -0,0 +1,1560 @@
...
+        ConvertIntToDecimalStringN(gStringVar1, 100, STR_CONV_MODE_LEFT_ALIGN, 3);
+        StringCopy(gStringVar2, gText_Gold);
+
+        gSpecialVar_Result = 0;
+
+        if (!VarGet(VAR_CURRENT_SECRET_BASE))
+            return;
+
+        VarSet(VAR_0x40EE, VarGet(VAR_0x40EE) | 0x10);
+
```
Comment (Diegoisawesome) [inline src/fldeff_80F9BCC.c]: Remove extra newlines

File: src/fldeff_80F9BCC.c [hunk 6]
```diff
@@ -0,0 +1,1560 @@
...
+{
+    s16 *data = gTasks[taskId].data;
+
+    switch (data[0])
+    {
+    case 0:
+        data[1] += 2;
+        if (data[1] > 8)
+            data[0]++;
+
```
Comment (Diegoisawesome) [inline src/fldeff_80F9BCC.c]: Remove extra newlines

**Comment by Diegoisawesome** [comment]
Also just in general, the file should probably be renamed to something better, like fldeff_misc.

File: src/fldeff_80F9BCC.c (hunk 2 above)
Comment (garakmon) [reply src/fldeff_80F9BCC.c]: :wesmart:

**Pull request merged**